pkg/network/ebpf/c/protocols/http/http.h

```diff
@@ -164,10 +164,13 @@ static __always_inline bool http_allow_packet(http_transaction_t *http, struct _
         return false;
     }
 
-    // if payload data is empty or if this is an encrypted packet, we only
-    // process it if the packet represents a TCP termination
-    bool empty_payload = skb_info->data_off == skb->len;
-    if (empty_payload || http->tup.sport == HTTPS_PORT || http->tup.dport == HTTPS_PORT) {
+    protocol_stack_t *stack = get_protocol_stack(&http->tup);
+    if (!stack) {
+        return false;
+    }
+    if (is_fully_classified(stack) || is_protocol_layer_known(stack, LAYER_ENCRYPTION)) {
+        // if this is an encrypted packet, we only
+        // process it if the packet represents a TCP termination
```

Comment:
why did you remove the old checks? especially empty payload

Reply:
it's more accurate to pass only tcp fin/rst for encrypted flow not based on port
I will re-add the empty payload check here